tests/testthat/test-cache.R

```diff
@@ -53,116 +53,92 @@ test_that("clear_cache deletes the directory if no file is given", {
 # `load_cache`
 
 test_that("load_cache loads the saved file in a new empty environment", {
-  with_mock(
-    `lintr::read_settings` = function(...) invisible(...),
-
-    e1 <- new.env(parent = emptyenv()),
-    assign("x", "foobar", envir = e1),
-    d1 <- tempfile(pattern = "lintr_cache_"),
-    f1 <- "R/test.R",
-    save_cache(cache = e1, file = f1, path = d1),
-    e2 <- load_cache(file = f1, path = d1),
-
-    expect_equal(ls(e2), "x"),
-    expect_equal(e2[["x"]], "foobar")
-  )
+  e1 <- new.env(parent = emptyenv())
+  assign("x", "foobar", envir = e1)
+  d1 <- tempfile(pattern = "lintr_cache_")
+  f1 <- "R/test.R"
+  save_cache(cache = e1, file = f1, path = d1)
+  e2 <- load_cache(file = f1, path = d1)
+
+  expect_equal(ls(e2), "x")
```

Comment:
Ideally this would just be `expect_equal(e2, e1)`

Reply:
Good to know that `expect_equal()` tolerates different environments as long as their contents match.

Reply:
not sure that's true -- hence using as.list() in the other test. I'll investigate further.